pyvrp/cpp/search/Route.h

```diff
@@ -985,32 +1013,32 @@ std::pair<Duration, Duration> Route::Proposal<Segments...>::duration() const
         auto ds = segment.duration(profile);
         auto first = segment.first();
 
-        if (first < data.numDepots())  // segment starts at depot
+        if (segment.startsAtReloadDepot())
             ds = ds.finaliseFront();
 
         auto const merge = [&](auto const &self, auto &&other, auto &&...args)
         {
             auto edgeDur = matrix(other.last(), first);
 
-            if (other.last() < data.numDepots())  // other ends at a depot
+            if (other.endsAtReloadDepot())
             {
-                // We can only finalise the current segment at the depot, so we
-                // need to travel there. The depot time windows restrictions are
-                // already handled by the other segment.
-                ds = DurationSegment::merge(edgeDur, {}, ds);
+                // The other segment ends at a reload depot, so we go there and
+                // finalise the current segment. We first travel there. We need
+                // to end the segment within the depot's time windows to
+                // properly account for any release time on our segment.
+                ProblemData::Depot const &depot = data.location(other.last());
+                ds = DurationSegment::merge(edgeDur, {depot}, ds);
```

Comment:
We need to finalise *after* we are at the depot, and in the depot's time windows. The way finalising works is that we put the artificial release time segment *before* the actual client/depot segment, so the depot needs to be in there, first.